### PR TITLE
Fix SDR compatibility with future PT 1.12

### PR DESCRIPTION
**Pull request by quancs**
## What does this PR do?

Fix the problem is disscussed in slack: SDR doesn't warn for low precision in PT 1.12

## Before submitting

- [x] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/metrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to **update the docs**?
- [ ] Did you write any new **necessary tests**?

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by Borda — APPROVED** [review]

**Comment by Borda** [comment]
well, looking at the parallel PR and seems it passes fine for PT 1.12 even without this fix: :/
https://github.com/PyTorchLightning/metrics/runs/5654852266?check_suite_focus=true

**Review by kaushikb11 — APPROVED** [review]

**Comment by quancs** [comment]
> well, looking at the parallel PR and seems it passes fine for PT 1.12 even without this fix: :/ https://github.com/PyTorchLightning/metrics/runs/5654852266?check_suite_focus=true

strange. in https://github.com/PyTorchLightning/metrics/runs/5654225685?check_suite_focus=true#step:7:3243, it didn't pass the test

File: tests/audio/test_sdr.py [hunk 1]
```diff
@@ -149,18 +149,22 @@ def test_on_real_audio():
...
-    ):
+
+    if _TORCH_GREATER_EQUAL_1_8 and _TORCH_LOWER_1_12:
+        with pytest.warns(
+            UserWarning,
+            match="Detected `nan` or `inf` value in computed metric, retrying computation in double precision",
+        ):
+            sdr_tm = signal_distortion_ratio(preds, target)
+    else:  # when pytorch < 1.8 or pytorch > 1.12, sdr doesn't have this problem
         sdr_tm = signal_distortion_ratio(preds, target)
```
Comment (quancs) [inline tests/audio/test_sdr.py:165]:
```suggestion
        sdr_tm = signal_distortion_ratio(preds, target).double()
```

**Comment by Borda** [comment]
could be that just the threshold is different and so with some mount randomness, it appears and sometimes not...

**Comment by quancs** [comment]
probablely

**Comment by Borda** [comment]
@quancs seems that now all test for PT 1.7 and lower is failing... if you could have look, pls :rabbit:

**Pull request merged**